### PR TITLE
Nested edits

### PR DESCRIPTION
**Pull request by jnwatson**
Implement nested/sub/conditional edits.

Improve schedGenr to avoid task cleanup issue.

### CODE REVIEW

File: synapse/lib/layer.py [hunk 1]
```diff
@@ -1234,18 +1240,33 @@ def getNodeTag(self, buid, tag):
         Args:
             nodeedit
 
-        Returns a (buid, form, edits) tuple.  edits contains the actual changes.
+        Returns a list of flattened post-NodeEdits (Tuple[BuidT, str, List]), with top-level and subedits all at the
```
[the remark this answers is not in the document]
Comment (jnwatson) [reply synapse/lib/layer.py:1243]: Correct. I'll fix the comment.

File: synapse/lib/base.py [hunk 2]
```diff
@@ -763,15 +764,16 @@ def __iter__(self):
             async for item in genr:
                 await q.put((True, item))
 
-            await q.put((False, None))
+            await q.put((False, s_common.NoValu))
```
[the remark this answers is not in the document]
Comment (jnwatson) [reply synapse/lib/base.py:767]: so you one yield None.

File: synapse/lib/layer.py [hunk 3]
```diff
@@ -1234,18 +1240,33 @@ def getNodeTag(self, buid, tag):
...
-        return changed
+            postedits.extend(changes)
+
+            if changes:
+                subedits = edit[2]
+                for ne in subedits:
+                    subpostnodeedits = await self._storNodeEdit(ne)
+                    # Consolidate any changes from subedits to the main edit node into that nodeedit's
+                    # edits, otherwise just append an entire nodeedit
+                    for subpostnodeedit in subpostnodeedits:
```
[the remark this answers is not in the document]
Comment (jnwatson) [reply synapse/lib/layer.py:1261]: Well, I tried it. It is about 8% slower.

File: synapse/lib/view.py [hunk 4]
```diff
@@ -286,8 +286,7 @@ def pack(self):
         if name not in ('name',):
             mesg = f'{name} is not a valid view info key'
             raise s_exc.BadOptValu(mesg=mesg)
-        #TODO when we can set more props, we may need to parse values.
-        await self.info.set(name, valu)
+        # TODO when we can set more props, we may need to parse values.  await self.info.set(name, valu)
```
Comment (invisig0th) [inline synapse/lib/view.py:289]: Was this an accident?

Comment (jnwatson) [reply synapse/lib/view.py]: good catch

File: synapse/lib/base.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jnwatson) [reply synapse/lib/base.py]: Discussed and resolved

**Pull request merged**